src/cluster.c

```diff
@@ -7090,9 +7090,9 @@ unsigned int delKeysInSlot(unsigned int hashslot) {
         robj *key = createStringObject(sdskey, sdslen(sdskey));
         dbDelete(&server.db[0], key);
         propagateDeletion(&server.db[0], key, server.lazyfree_lazy_server_del);
-        propagatePendingCommands();
         signalModifiedKey(NULL, &server.db[0], key);
         moduleNotifyKeyspaceEvent(NOTIFY_GENERIC, "del", key, server.db[0].id);
+        afterDatasetChange();
```

Comment:
please document this bugfix in the top comment

Reply:
As discussed with @oranagra I will remove this fix from here and will do it on separate PR.